services/advisoryValidator.js

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable no-empty-function */
+/* eslint-disable no-unused-vars */
+import { DOMParser } from '@xmldom/xmldom';
+import xpath from 'xpath';
+import { getAdvisoryDefinitions } from '../utils/utils.js';
+import { checkIfIatiIdentifiersInDatastore } from '../utils/datastoreServices.js';
+
+const select = xpath.useNamespaces({ xml: 'http://www.w3.org/XML/1998/namespace' });
+
+
+function groupListIntoMap(list, keyGetter) {
```

Comment:
I've not done this in a while but is there a way to simplify this function with a reduce or groupBy?

Reply:
Yeah, the `groupBy` requires an extra import which I wanted to avoid, and I wasn't sure about the readability of `reduce` since it's not really reducing but grouping, but on second look it is quite nice and simple using `reduce`, and it saves introducing that utility function, so is best overall I think.